library/core/src/iter/mod.rs

```diff
@@ -352,6 +352,31 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
+#[doc(hidden)]
+#[unstable(feature = "impl_fold_via_try_fold", reason = "internal use only", issue = "none")]
+#[macro_export]
```

Comment:
Do you plan to use the macro outside of the `iter` module? Because if no, you can remove this macro_export, as all uses are inside the module (and the macro is declared before the mod statements). Then you also don't need the `__` or the unstable feature.

Reply:
You're right, I should try harder at this.  I did this originally because `core::iter::impl_fold_via_try_fold!` wasn't working for me, but I think that was me forgetting about the order-dependence of macro compilation.  I can probably do better, and worry about moving it outside `iter` later only if needed.  (`NeverShortCircuit` is currently `pub(crate)` anyway, so it definitely doesn't need exporting.)

@rustbot author

Reply:
Thanks for prodding, est31, it worked great 👍 

@rustbot ready